tests/sources/test_redis.py

```diff
@@ -0,0 +1,370 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+import json
+from contextlib import asynccontextmanager
+from unittest import mock
+from unittest.mock import ANY, AsyncMock, Mock
+
+import pytest
+import redis
+from freezegun import freeze_time
+
+from connectors.filtering.validation import SyncRuleValidationResult
+from connectors.protocol import Filter
+from connectors.source import ConfigurableFieldValueError
+from connectors.sources.redis import (
+    RedisAdvancedRulesValidator,
+    RedisDataSource,
+)
+from tests.commons import AsyncIterator
+from tests.sources.support import create_source
+
+ADVANCED_SNIPPET = "advanced_snippet"
+
+DOCUMENT = [
```

Comment:
```suggestion
DOCUMENTS = [
```

As this is an array I would use plural here